pdns/packethandler.cc

```diff
@@ -438,13 +438,51 @@ bool PacketHandler::getBestWildcard(DNSPacket& p, const SOAData& sd, const DNSNa
   return haveSomething;
 }
 
+DNSName PacketHandler::doAdditionalServiceProcessing(const DNSName &firstTarget, const uint16_t &qtype, const int domain_id, std::unique_ptr<DNSPacket>& r) {
+  DNSName ret = firstTarget;
+  size_t ctr = 5; // Max 5 SVCB Aliasforms per query
+  bool done = false;
+  while (!done && ctr > 0) {
+    DNSZoneRecord rr;
+    done = true;
+    B.lookup(QType(qtype), ret, domain_id);
+    while (B.get(rr)) {
+      rr.dr.d_place = DNSResourceRecord::ADDITIONAL;
+      switch (qtype) {
+        case QType::SVCB: {
+          auto rrc = getRR<SVCBRecordContent>(rr.dr);
+          r->addRecord(std::move(rr));
+          ret = rrc->getTarget().isRoot() ? ret : rrc->getTarget();
+          if (rrc->getPriority() == 0) {
+            done = false;
+          }
+          break;
+        }
+        case QType::HTTPS: {
+          auto rrc = getRR<HTTPSRecordContent>(rr.dr);
+          r->addRecord(std::move(rr));
+          ret = rrc->getTarget().isRoot() ? ret : rrc->getTarget();
+          if (rrc->getPriority() == 0) {
+            done = false;
+          }
+          break;
+        }
+        default:
+          abort();
```

Comment:
That sounds a bit harsh?

Reply:
it is a programmer error if it ever triggered, but as @zeha suggests, I'll make it a PDNSException